app/components/questions/BelongingRates.js

```diff
@@ -156,7 +166,25 @@ function AreaSelector(props) {
         ))}
       </select>
     </div>
-  )
+  );
+}
+
+function GenerateMessage(props) {
+  if(!props.areaData){
+    return (
+      <p>Waiting for data...</p>  
+    )
+  }
+
+  if(props.dataFlag){
```

Comment:
Instead of putting this in data flag, we could put the check here.

Also, instead of checking:

```js
areaData.belongingRateFormatted==="NaN%"
```

We can check the non-formatted field:

```
!areaData.belonging_rate_2018
```

Note the not `!` which will check if the value is null or undefined. This will also exclude `0`, which might be a legitimate value, so if you want to cover that case you can do:

```js
const missingData = !areaData.disability_rate_2018 && areaData.disability_rate_2018 !== 0
if (missingData) ...
```

You can also "destructure" the props that way you don't have to do `prop.*` everytime:

```js
const { areaData } = props; 
```

Or:

```js
function GenerateMessage({ areaData }) {
```